### ISSUE TITLE
Streamlabs monitoring Issue

### ISSUE THREAD

**Issue by Shnikes0731** [ExistentialAudio/BlackHole#321]
HI! This was working last weekend but as soon as I logged back on to my Apple M1 MacBook Pro I was having issues. If I use the multi output device it cuts my audio from my speakers or headphones. Right now I have the Blackhole 2ch selected (not in the multi output) as my audio output in my MIDI and it's playing perfectly through my stream(and not through speakers/headphones because it's not selected of course). But as soon as I add a multi-output and check the Blackhole box (I make sure that its below my built in audio and my headphones) it cuts audio from my stream AND speakers/headphones. It's a little strange because this worked perfectly first try last weekend. I already tried uninstalling and reinstalling, restarting streamlabs, and I've restarted my computer as well. I feel like I'm missing just one small detail. Any help is appreciated! Thanks so much.

**Comment by devinroth** [comment]
Sounds like audio drift.

Make sure you have drift correction enabled.

Devin

**Comment by Shnikes0731** [comment]
> Sounds like audio drift. Make sure you have drift correction enabled. Devin
> […](#)

I have drift enabled on all the channels just to try and it didn't change much. It seemed like it made the delay of audio between my speakers and headphones less drastic but other than that I still can only hear audio on stream or in my headphones. Not both at once. I don't know if it's allowed but if you want to hop in the stream I'm trying to figure it out live. Not self promoting just let me know if that is an option for you. Thanks for responding, you a real one. (I'm on twitch.tv/itsdevynwithawhy) Just delete this if I'm not allowed to post that. Again, thank you!

**Comment by devinroth** [comment]
Upload a screenshot.

Devin

**Comment by Shnikes0731** [comment]
Okay so here I'm showing that the stream is picking up game audio with the mixer channel "Blackhole 2ch" on the left and on the right is the MIDI with all of the things activated the way that I thought it would be and the way I did it last weekend (minus the drift suggestion).
<img width="1440" alt="Screen Shot 2021-02-26 at 2 44 58 PM" src="https://user-images.githubusercontent.com/79717922/109347489-402f9680-7841-11eb-99ca-841842622c10.png">

**Comment by devinroth** [comment]
Try putting your MacBook Pro Speakers on the top of the multi.

Devin

**Comment by Shnikes0731** [comment]
That didn't change anything unfortunately. I'm sending these to hopefully clarify the situation more maybe? Not sure what to do at this point just trying to help as much as I can.

 So this screenshot shows the configuration of audio output sources selected for my headphones to hear game audio:
<img width="1440" alt="Screen Shot 2021-02-26 at 3 13 59 PM" src="https://user-images.githubusercontent.com/79717922/109350378-66573580-7845-11eb-8a6c-bfd9f5087c3e.png">

This screenshot is the configuration for not the stream nor I hearing any game audio at all:
<img width="1440" alt="Screen Shot 2021-02-26 at 3 14 08 PM" src="https://user-images.githubusercontent.com/79717922/109350383-68b98f80-7845-11eb-93d1-d8cb7d5c48ca.png">

And this screen shot is when I select the BlackHole 2ch output individually (not the multi output device)and only the stream can hear game audio:
<img width="1440" alt="Screen Shot 2021-02-26 at 3 14 23 PM" src="https://user-images.githubusercontent.com/79717922/109350389-6b1be980-7845-11eb-9b02-b858bebbc45a.png">

**Comment by devinroth** [comment]
Do a test with audacity. Can you hear and record?

Devin

**Comment by Shnikes0731** [comment]
I just captured this through streamlabs itself. I couldn't hear anything through my headphones but its coming through stream right now as well as through the capture.
https://user-images.githubusercontent.com/79717922/109352051-3bbaac00-7848-11eb-8c4b-70496669336a.mp4

**Comment by devinroth** [comment]
Grab a screenshot of blackHole in Activity monitor.

Devin

**Comment by Shnikes0731** [comment]
I wasn't sure which one you were looking for so I took these 3. I do have to get going for a few hours but I'll be back attempting to figure out the issue later tonight in about 5 hours or so (9pm EST). Thank you for helping out and responding to me so far. I'm super grateful.

<img width="1440" alt="Screen Shot 2021-02-26 at 3 45 24 PM" src="https://user-images.githubusercontent.com/79717922/109353087-dcf63200-7849-11eb-9438-cae18c4259c8.png">
<img width="1440" alt="Screen Shot 2021-02-26 at 3 45 45 PM" src="https://user-images.githubusercontent.com/79717922/109353091-de275f00-7849-11eb-997f-6d27b73902cd.png">
<img width="1440" alt="Screen Shot 2021-02-26 at 3 46 21 PM" src="https://user-images.githubusercontent.com/79717922/109353094-df588c00-7849-11eb-8ad2-eee64474b243.png">

**Comment by devinroth** [comment]
I'm listen to audio with a similar setup on a M1 Mac mini without any issues..

Not sure where things are going wrong for you. 

<img width="763" alt="Screen Shot 2021-02-26 at 12 55 42 PM" src="https://user-images.githubusercontent.com/13010961/109354076-17ec6b80-7832-11eb-9d43-325da50a550d.png">

**Comment by Shnikes0731** [comment]
> I'm listen to audio with a similar setup on a M1 Mac mini without any issues..
> 
> Not sure where things are going wrong for you.
> 
> <img alt="Screen Shot 2021-02-26 at 12 55 42 PM" width="763" src="https://user-images.githubusercontent.com/13010961/109354076-17ec6b80-7832-11eb-9d43-325da50a550d.png">

Yeah I know it’s super weird. I did everything the same way last Sunday and now it just doesn’t work. It’s like as soon as I add the BlackHole to the multi-output it interferes with something else or something like that. It’s pretty strange. Any ideas on how I can just restart the process cleanly? I’m open to anything at this point.

**Comment by devinroth** [comment]
Restart.

But you’ve already done that.

Multi output have always been buggy. #1 cause of problems. I should just write my own.

Devin

**Comment by Shnikes0731** [comment]
> Restart. But you’ve already done that. Multi output have always been buggy. #1 cause of problems. I should just write my own. Devin
> […](#)
Hmmm... Gotcha. I'll just keep trying I suppose. Not much else I can do, right?

**Comment by devinroth** [comment]
I just had an idea.

Some apps don’t support multi-outputs. Are you using the same game as last time?

If you play something from YouTube does the multi-output work as expected. If so, the app is the issue and you’ll need to find a work around.

Devin

**Comment by Shnikes0731** [comment]
Not really. It’s every audio from my computer except the notification sounds that don’t play through my computer or headphones. It definitely feels like a work around has to be found but I don’t even know where to start.

**Comment by devinroth** [comment]
Does it work normally when playing audio through YouTube?

Devin

**Comment by Shnikes0731** [comment]
It either plays through the stream or through my headphones. Never both at once

**Comment by devinroth** [comment]
Hahah I know what the problem is. You created an aggregate instead of a multi output.

Delete it and create a new one.

Devin

**Comment by Shnikes0731** [comment]
I haven't used the aggregate device option yet I don't think. If you look at my other screenshots it shows I was using multi-output

**Comment by devinroth** [comment]
Trust me.

Devin

**Comment by devinroth** [comment]
The device call multi output is not a multi output.

Proof in your photo. It’s the wrong icon.

Devin

**Comment by devinroth** [comment]
And it completely explains your problem.

Devin

**Comment by devinroth** [comment]
Maybe I’m wrong.

Worth trying.

Devin

**Comment by devinroth** [comment]
Does it work as expected without using BlackHole in your multi-output?

**Comment by LeonardCA** [comment]
Consider checking the streamlabs mixer settings and make sure that the output for your Aggregate Output device is set to "monitor and output".  Audio on a mac is a house of cards.  After changing these settings, sometimes it's necessary to restart everything.
<img width="522" alt="2021-03-11 AudioSettingLocation" src="https://user-images.githubusercontent.com/80489739/110866671-5c3c2a80-827a-11eb-995e-4738b1f26b63.png">
<img width="833" alt="2021-03-11 Advanced AudioSettings" src="https://user-images.githubusercontent.com/80489739/110866710-6ceca080-827a-11eb-96f5-952770b62ce9.png">

**Comment by orchetect** [comment]
Does this issue persist with the latest 0.3.0 release?
